core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1493,6 +1493,7 @@ class KafkaConfig private(doLog: Boolean, val props: java.util.Map[_, _], dynami
 
   // Cache the current config to avoid acquiring read lock to access from dynamicConfig
   @volatile private var currentConfig = this
+  val processRoles: Set[ProcessRole] = parseProcessRoles()
```

Comment:
hmm, what's the reason to move this?

Reply:
On the line below, we are creating the DynamicBrokerConfig which gets a partially initialized KafkaConfig. Moving `parseProcessRoles` up here was simpler than refactoring a bunch of this config code :)

Reply:
thanks for the explanation. sounds good